q2_sample_classifier/classify.py

```diff
@@ -42,13 +42,16 @@ def metatable(ctx,
               metadata,
               table=None,
               missing_samples='ignore',
-              missing_values='error'):
+              missing_values='error',
+              drop_all_unique=False):
     # gather numeric metadata
     metadata = metadata.filter_columns(
-        column_type='numeric', drop_all_unique=True, drop_zero_variance=True,
-        drop_all_missing=True).to_dataframe()
+        column_type='numeric', drop_all_unique=drop_all_unique,
+        drop_zero_variance=True, drop_all_missing=True).to_dataframe()
     # drop columns with negative values
-    metadata = metadata[-(metadata < 0).any(axis=1)]
+    # pd.DataFrame.any() reverses the normal axis definition: axis=0 reduces
+    # the index, return a Series whose index is the original column labels.
+    metadata = metadata.loc[:, -(metadata < 0).any(axis=0)]
```

Comment:
👍

Reply:
Oh, I wasn't paying close attention when I commented the first time, if you go with the second option, you'll need to change any() to all(). I'm not sure the second option is actually cheaper as a result.

Reply:
yep I saw that — not cheaper, but definitely easier to read (I think what I had before was the python equivalent of a double-negative)

seems to have broken the tests but I will work through that now...

Reply:
oh I think why I had it so convoluted before is because `metadata.loc[:, (metadata >= 0).all(axis=0)]` would also drop any columns containing nans; we don't want to do this because we want to let `dropna` choose the axis to drop nans along in the next step

simple solution: do negative column value filtering _after_ dropna.

Reply:
Ah, that's a clever way to preserve the NaNs! Yeah just moving it would be easier to read.